README.md

# ACWR

<!-- badges: start -->

[![License:
MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)
<!-- badges: end -->

## Overview

The **ACWR** package have been designed to calculate the the acute
chronic workload ratio using three different methods: exponentially
weighted moving average (EWMA), rolling average coupled (RAC) and
rolling averaged uncoupled (RAU).

## Resources

  - [Application of ACWR](https://bjsm.bmj.com/content/50/5/273)
    (European Journal of Sport Science)
  - [Rolling average method](https://bjsm.bmj.com/content/53/16/988)
    (British Journal of Sports Medicine)
  - [Exponentially weighted moving averages
    method](https://bjsm.bmj.com/content/51/9/749.long) (British Journal
    of Sports Medicine)

## Example

This is a basic example which shows you how to use the ACWR package:

``` r
library(devtools)
install_github("JorgeDelro/ACWR")
library(ACWR)
```

First, we have to load the data stored in the package

``` r
data("training_load", package = "ACWR")
# Convert to data.frame
training_load <- data.frame(training_load)
```

Then, we can calculate the ACWR:

``` r
result_ACWR <- ACWR(db = training_load,
                  ID = "ID",
                  TL = "TL",
                  weeks = "Week",
                  training_dates = "Training_Date",
                  ACWR_method = c("EWMA", "RAC", "RAU"))
```

Additionally, individual plot can be obtained:

``` r
ACWR_plot <- plot_ACWR(db = result_ACWR,
                        TL = "TL",
                        ACWR = "RAC_ACWR",
                        day = "Day",
                        ID = "ID")
```

Functions for each individual method have been implemented too:

``` r
# Select the first subject
training_load_1 <- training_load[training_load[["ID"]] == 1,  ]

# EWMA
result_EWMA <- EWMA(TL = training_load_1$TL)

# RAC
result_RAC <- RAC(TL = training_load_1$TL,
                  weeks = training_load_1$Week,
                  training_dates = training_load_1$Training_Date)
                    
# RAU
result_RAU <- RAU(TL = training_load_db_1$TL,
                  weeks = training_load_1$Week,
                  training_dates = training_load_1$Training_Date)
```


NEWS.md
# ACWR 0.0.0.9000

* Added a `NEWS.md` file to track changes to the package.


tests/testthat.R
library(testthat)
library(ACWR)

test_check("ACWR")


tests/testthat/test-RAC.R

test_that("output of RAC function", {

  # Get old working directory
  oldwd <- getwd()

  # Set temporary directory
  setwd(tempdir())

  # Simulate Training load
  Week <- c(
    rep(x = 1, n = 7),
    rep(x = 2, n = 7),
    rep(x = 3, n = 7),
    rep(x = 4, n = 7)
  )
  TL <- c(
    rep(x = 30, times = 21),
    rep(x = 50, times = 7)
  )
  Training_Date <- seq(as.Date("2022/1/1"), as.Date("2022/1/28"), "days")

  training_data <- data.frame(Week, TL, Training_Date)

  result_RAC <- RAC(TL = training_data$TL,
                      weeks = training_data$Week,
                      training_dates = training_data$Training_Date)

  expect_equal(is.list(result_RAC), TRUE)
  expect_equal(names(result_RAC)[1], "RAC_acute")
  expect_equal(names(result_RAC)[2], "RAC_chronic")
  expect_equal(names(result_RAC)[3], "RAC_ACWR")

  # set user working directory
  setwd(oldwd)

})


tests/testthat/test-RAU.R

test_that("output of RAU function", {

  # Get old working directory
  oldwd <- getwd()

  # Set temporary directory
  setwd(tempdir())

  # Simulate Training load
  Week <- c(
    rep(x = 1, n = 7),
    rep(x = 2, n = 7),
    rep(x = 3, n = 7),
    rep(x = 4, n = 7)
  )
  TL <- c(
    rep(x = 30, times = 21),
    rep(x = 50, times = 7)
  )
  Training_Date <- seq(as.Date("2022/1/1"), as.Date("2022/1/28"), "days")

  training_data <- data.frame(Week, TL, Training_Date)

  result_RAU <- RAU(TL = training_data$TL,
                    weeks = training_data$Week,
                    training_dates = training_data$Training_Date)

  expect_equal(is.list(result_RAU), TRUE)
  expect_equal(names(result_RAU)[1], "RAU_acute")
  expect_equal(names(result_RAU)[2], "RAU_chronic")
  expect_equal(names(result_RAU)[3], "RAU_ACWR")

  # set user working directory
  setwd(oldwd)

})


R/RAC.R

#' Rolling Average Coupled
#'
#' @param TL training load
#' @param weeks training weeks
#' @param training_dates training dates
#'
#' @return {This function returns the following variables:
#' \itemize{
#' \item RAC_chronic: RAC - chronic training load.
#' \item RAC_acute: RAC - acute training load.
#' \item RAC_ACWR: RAC - Acute-Chronic Workload Ratio.
#' }}
#' @export
#'
#' @examples
#'
#' \dontrun{
#' # Get old working directory
#' oldwd <- getwd()
#'
#' # Set temporary directory
#' setwd(tempdir())
#'
#' # Read db
#' data("training_load", package = "ACWR")
#'
#' # Convert to data.frame
#' training_load <- data.frame(training_load)
#'
#' # Select the first subject
#' training_load_1 <- training_load[training_load[["ID"]] == 1,  ]
#'
#' # Calculate ACWR
#' result_RAC <- RAC(TL = training_load_1$TL,
#'                    weeks = training_load_1$Week,
#'                    training_dates = training_load_1$Training_Date)
#'
#' # set user working directory
#' setwd(oldwd)
#' }
#'
RAC <- function(TL,
                weeks,
                training_dates) {

  # Count number of sessions / week
  sessions_week <- as.data.frame(table(weeks))

  # Initialize variables
  RAC_chronic <- c()
  RAC_acute <- c()
  RAC_ACWR <- c()
  # Initialize number of training sessions
  n_sessions_total <- 0

  # Loop over the total days of training
  for (i in unique(weeks)) {

    # First training week: RAC_chronic = RAC_acute
    if(i == 1){

      # loop over number of sessions / week
      for (j in 1:sessions_week$Freq[unique(weeks)[i]]) {

        # First training day: RAC_chronic = TL / RAC_acute = TL
        if(j == 1){
          # Count number of training sessions
          n_sessions_total <- n_sessions_total + 1
          RAC_chronic[n_sessions_total] = TL[n_sessions_total]
          RAC_acute[n_sessions_total] = TL[n_sessions_total]
        }

        # Rest of the week
        else if(j >= 2){
          # Count number of training sessions
          n_sessions_total <- n_sessions_total + 1
          RAC_chronic[n_sessions_total] = (sum(TL[1:n_sessions_total]))/n_sessions_total
          RAC_acute[n_sessions_total] = (sum(TL[1:n_sessions_total]))/n_sessions_total
        }

      }
    } # end first week

    # from second week to end of first month
    else if(i >= 2 && i < 5){

      # loop over number of sessions / week
      for (j in 1:sessions_week$Freq[unique(weeks)[i]]) {

        # Count number of training sessions
        n_sessions_total <- n_sessions_total + 1
        RAC_chronic[n_sessions_total] = (sum(TL[1:n_sessions_total]))/n_sessions_total

        # RAC acute each 7 CALENDAR days
        # Calculate 7 days training blocks
        # Returns:
        # n_sessions_acute = Number of training sessions include in the acute block
        # previous_TL_acute = Position of the first session of the acute training block
        acute_TB <- training_blocks(training_dates = training_dates,
                                    actual_TL = n_sessions_total,
                                    diff_dates = 6)

        RAC_acute[n_sessions_total] = (sum(TL[acute_TB$previous_TL:n_sessions_total]))/acute_TB$n_sessions
      }

    } # end first month

    # from second moth to end of data
    else if(i >= 5){

      # loop over number of sessions / week
      for (j in 1:sessions_week$Freq[unique(weeks)[i]]) {
        # Count number of training sessions
        n_sessions_total <- n_sessions_total + 1

        # RAC chronic each 28 CALENDAR days
        # Calculate 28 days training blocks
        chronic_TB <- training_blocks(training_dates = training_dates,
                                      actual_TL = n_sessions_total,
                                      diff_dates = 27)
        RAC_chronic[n_sessions_total] = (sum(TL[chronic_TB$previous_TL:n_sessions_total]))/chronic_TB$n_sessions

        # RAC acute each 7 CALENDAR days
        # Calculate 7 days training blocks
        acute_TB <- training_blocks(training_dates = training_dates,
                                    actual_TL = n_sessions_total,
                                    diff_dates = 6)
        RAC_acute[n_sessions_total] = (sum(TL[acute_TB$previous_TL:n_sessions_total]))/acute_TB$n_sessions
      }

    } # end >= second moth

  } # end of loop over the data

  # Calculate ACWR
  RAC_ACWR <- RAC_acute / RAC_chronic

  return(list(RAC_acute = round(RAC_acute, 2),
              RAC_chronic = round(RAC_chronic, 2),
              RAC_ACWR = round(RAC_ACWR, 2)))

}# end RAC function


R/RAU.R

#' Rolling Average Uncoupled
#'
#' @param TL training load
#' @param weeks training weeks
#' @param training_dates training dates
#'
#' @return {This function returns the following variables:
#' \itemize{
#' \item RAU_chronic: RAU - chronic training load.
#' \item RAU_acute: RAU - acute training load.
#' \item RAU_ACWR: RAU - Acute-Chronic Workload Ratio.
#' }}
#' @export
#'
#' @examples
#'
#' \dontrun{
#' # Get old working directory
#' oldwd <- getwd()
#'
#' # Set temporary directory
#' setwd(tempdir())
#'
#' # Read db
#' data("training_load", package = "ACWR")
#'
#' # Convert to data.frame
#' training_load <- data.frame(training_load)
#'
#' # Select the first subject
#' training_load_1 <- training_load[training_load[["ID"]] == 1,  ]
#'
#' # Calculate ACWR
#' result_RAU <- RAU(TL = training_load_1$TL,
#'                    weeks = training_load_1$Week,
#'                    training_dates = training_load_1$Training_Date)
#'
#' # set user working directory
#' setwd(oldwd)
#' }
#'
RAU <- function (TL,
                 weeks,
                 training_dates) {

  # Count number of sessions / week
  sessions_week <- as.data.frame(table(weeks))

  # Initialize variables
  RAU_chronic <- c()
  RAU_acute <- c()
  RAU_ACWR <- c()
  # Initialize number of training sessions
  n_sessions_total <- 0
  # We also need a new counter for the number of training sessions
  n_sessions_chronic <- 1

  # Loop over the total days of training
  for (i in unique(weeks)) {

    # First training week: RAC_chronic = NA / RAC_acute = Training load
    if(i == 1){

      # loop over number of sessions / week
      for (j in 1:sessions_week$Freq[unique(weeks)[i]]) {

        # First training day: RAC_chronic = TL / RAC_acute = TL
        if(j == 1){
          # Count number of training sessions
          n_sessions_total <- n_sessions_total + 1
          # RAU_chronic[n_sessions_total] = TL[n_sessions_total]
          RAU_chronic[n_sessions_total] = NA
          RAU_acute[n_sessions_total] = TL[n_sessions_total]
        }

        # Rest of the week
        else if(j >= 2){
          # Count number of training sessions
          n_sessions_total <- n_sessions_total + 1
          # RAU_chronic[n_sessions_total] = (sum(TL[1:n_sessions_total]))/n_sessions_total
          RAU_chronic[n_sessions_total] = NA
          RAU_acute[n_sessions_total] = (sum(TL[1:n_sessions_total]))/n_sessions_total
        }

        # During first week of RAU ACWR = 0
        RAU_ACWR[n_sessions_total] <- NA

      }
    } # end first week

    # from the beginning of the second week to end of third week
    else if(i >= 2 && i < 5){

      # loop over number of sessions / week
      for (j in 1:sessions_week$Freq[unique(weeks)[i]]) {

        # Count number of training sessions
        n_sessions_total <- n_sessions_total + 1

        # RAU acute each 7 CALENDAR days
        # Calculate 7 days training blocks
        # Returns:
        # n_sessions = Number of training sessions include in the acute block
        # previous_TL = Position of the first session of the acute training block
        acute_TB <- training_blocks(training_dates = training_dates,
                                    actual_TL = n_sessions_total,
                                    diff_dates = 6)

        RAU_acute[n_sessions_total] = (sum(TL[acute_TB$previous_TL:n_sessions_total]))/acute_TB$n_sessions


        # RAU chronic
        # (acute_TB$previous_TL-1) indicates the position of the first session
        # We are going to reuse this value to indicate the first value of the RAU chronic block
        RAU_chronic[n_sessions_total] = (sum(TL[(acute_TB$previous_TL-1):1]))/n_sessions_chronic
        n_sessions_chronic <- n_sessions_chronic + 1
      }
    } # end first 3 weeks

    # from fourth week to end of data
    else if(i >= 5){

      # loop over number of sessions / week
      for (j in 1:sessions_week$Freq[unique(weeks)[i]]) {

        # Count number of training sessions
        n_sessions_total <- n_sessions_total + 1

        # RAU acute each 7 CALENDAR days
        # Calculate 7 days training blocks
        # Returns:
        # n_sessions = Number of training sessions include in the acute block
        # previous_TL = Position of the first session of the acute training block
        acute_TB <- training_blocks(training_dates = training_dates,
                                    actual_TL = n_sessions_total,
                                    diff_dates = 6)

        RAU_acute[n_sessions_total] = (sum(TL[acute_TB$previous_TL:n_sessions_total]))/acute_TB$n_sessions


        # RAU chronic
        #
        chronic_TB <- training_blocks(training_dates = training_dates,
                                      actual_TL = n_sessions_total,
                                      diff_dates = 20)
        # Number of sessions include in the chronic training block =
        # Number of sessions in chronic - number of sessions in acute
        RAU_chronic[n_sessions_total] = (sum(TL[acute_TB$previous_TL:chronic_TB$previous_TL]))/chronic_TB$n_session
      }


    }
  } # end loop

  # Calculate ACWR
  RAU_ACWR <- RAU_acute / RAU_chronic

  return(list(RAU_acute = round(RAU_acute, 2),
              RAU_chronic = round(RAU_chronic, 2),
              RAU_ACWR = round(RAU_ACWR, 2)))

} # end RAU function


R/ACWR.R

#' Acute Chronic Workload Ratio
#'
#' @param db a data frame
#' @param ID ID of the subjects
#' @param TL training load
#' @param weeks training weeks
#' @param days training days
#' @param training_dates training dates
#' @param ACWR_method method to calculate ACWR
#'
#' @return a data frame with the acute & chronic training load and ACWR calculated
#'         with the selected method/s and added on the left side of the data frame

#' @export
#'
#' @examples
#'
#' \dontrun{
#' # Get old working directory
#' oldwd <- getwd()
#'
#' # Set temporary directory
#' setwd(tempdir())
#'
#' # Read dfs
#' data("training_load", package = "ACWR")
#'
#' # Convert to data.frame
#' training_load <- data.frame(training_load)
#'
#' # Calculate ACWR
#' result_ACWR <- ACWR(db = training_load,
#'                  ID = "ID",
#'                  TL = "TL",
#'                  weeks = "Week",
#'                  days = "Day",
#'                  training_dates = "Training_Date",
#'                  ACWR_method = c("EWMA", "RAC", "RAU"))
#'
#' # set user working directory
#' setwd(oldwd)
#' }
#'
ACWR <- function(db,
                 ID,
                 TL,
                 weeks,
                 days,
                 training_dates,
                 ACWR_method = c("EWMA", "RAC", "RAU")) {

  ## Checks
  # NULL
  if(is.null(db)){
    stop("you must provide a dataframe")
  }
  if(is.null(ID)){
    stop("you must provide the column name of ID variable")
  }
  if(is.null(TL)){
    stop("you must provide the column name of training load variable")
  }
  if(is.null(weeks)){
    stop("you must provide the column name of week variable")
  }
  if(is.null(days)){
    stop("you must provide the column name of day variable")
  }
  if(is.null(training_dates)){
    stop("you must provide the column name of training dates variable")
  }

  # Loop over the subjects
  for (i in unique(db[[ID]])) {

    # Create individual dfs
    db_ind <- db[db[[ID]] == i,  c(ID, TL, weeks, days, training_dates) ]

    # Loop over the methods
    for (j in ACWR_method) {

      # EWMA method
      if(j == "EWMA") {

        res_EWMA <- EWMA(TL = db_ind[[TL]])

        db_ind$EWMA_chronic <- res_EWMA$EWMA_chronic
        db_ind$EWMA_acute <- res_EWMA$EWMA_acute
        db_ind$EWMA_ACWR <- res_EWMA$EWMA_ACWR

      }

      # Rolling Average Coupled
      if(j == "RAC") {

        res_RAC <- RAC(TL = db_ind[[TL]],
                       weeks = db_ind[[weeks]],
                       training_dates = db_ind[[training_dates]])

        db_ind$RAC_acute = res_RAC$RAC_acute
        db_ind$RAC_chronic = res_RAC$RAC_chronic
        db_ind$RAC_ACWR = res_RAC$RAC_ACWR

      }

      # Rolling Average Uncoupled
      if(j == "RAU") {

        res_RAU <- RAU(TL = db_ind[[TL]],
                       weeks = db_ind[[weeks]],
                       training_dates = db_ind[[training_dates]])

        db_ind$RAU_acute = res_RAU$RAU_acute
        db_ind$RAU_chronic = res_RAU$RAU_chronic
        db_ind$RAU_ACWR = res_RAU$RAU_ACWR

      }

    } # end loop over methods

    if(i == 1){
      db_final <- db_ind
    } else {
      db_final <- rbind(db_final, db_ind)
    }

  } # end loop over individuals

  return(db_final)

} # end ACWR function


R/EWMA.R

#' Exponentially Weighted Moving Average
#'
#' @param TL training load
#'
#' @return {This function returns the following variables:
#' \itemize{
#' \item EWMA_chronic: EWMA - chronic training load.
#' \item EWMA_acute: EWMA - acute training load.
#' \item EWMA_ACWR: EWMA - Acute-Chronic Workload Ratio.
#' }}
#'
#' @export
#'
#' @examples
#'
#' \dontrun{
#' # Get old working directory
#' oldwd <- getwd()
#'
#' # Set temporary directory
#' setwd(tempdir())
#'
#' # Read db
#' data("training_load", package = "ACWR")
#'
#' # Convert to data.frame
#' training_load <- data.frame(training_load)
#'
#' # Select the first subject
#' training_load_1 <- training_load[training_load[["ID"]] == 1,  ]
#'
#' # Calculate ACWR
#' result_EWMA <- EWMA(TL = training_load_1$TL)
#'
#' # set user working directory
#' setwd(oldwd)
#' }
#'
EWMA <- function(TL) {


  # lambda <- 2/(N + 1)

  # Initialize variables
  EWMA_chronic <- c()
  EWMA_acute <- c()
  EWMA_ACWR <- c()
  lambda_acute <- 2/(7+1)
  lambda_chronic <- 2/(28+1)

  # Loop over the TL
  for (i in seq_along(TL)) {

    # First training day: EWMA_chronic = TL / EWMA_acute = TL
    if(i == 1){
      EWMA_chronic[i] = TL[i]
      EWMA_acute[i] = TL[i]
    }

    if(i > 1){
      EWMA_chronic[i] = TL[i] * lambda_chronic + ((1- lambda_chronic)* EWMA_chronic[i-1])
      EWMA_acute[i] = TL[i] * lambda_acute + ((1- lambda_acute)* EWMA_acute[i-1])
    }

    EWMA_ACWR <- EWMA_acute / EWMA_chronic

  }

  return(list(EWMA_acute = EWMA_acute,
              EWMA_chronic = EWMA_chronic,
              EWMA_ACWR = EWMA_ACWR))

} # end EWMA


R/utils.R

#' Create Training Blocks
#'
#' @param training_dates training dates
#' @param actual_TL position of the actual training load
#' @param diff_dates difference in days
#'
#'
training_blocks <- function(training_dates,
                            actual_TL,
                            diff_dates
){

  # Initialize variables
  n_sessions <- 0

  # Create blocks BACKWARDS

  # loop over the training days backwards, starting from actual training day - 1
  for (previous_TL in rev(1:actual_TL-1)) {

    # Count total number of sessions include in a acute training block
    n_sessions <- n_sessions + 1

    # Calculate the difference in days between 2 dates (as integer)
    diff_dates_calculated <- as.integer(training_dates[actual_TL] - training_dates[previous_TL])

    # If the difference between 2 dates are X days or more more then stop the loop
    if(diff_dates_calculated >= diff_dates) {
      break
    }


  } # end loop

  # Number of training sessions include in the training block
  return(list(n_sessions = n_sessions +1,
              # Position of the first session of the training block
              previous_TL = previous_TL)
  )


} # end training blocks


R/ACWR_plot.js

//Set initial values
var margin = options.margin,
    width = width-(2*margin),
    height = height-(2*margin),
    y0max = d3.max(data, function(d) { return d.TL; });
    y1max = d3.max(data, function(d) { return d.ACWR; });

// Range of x-axis values
var x = d3.scaleBand()
    .domain(data.map(d => d.day))
    .range([margin, margin+width])
    .paddingInner(0.3)
    .paddingOuter(0.3);

    svg.append("g")
      .attr("transform", "translate(" + 0 + "," + (height+margin) + ")")
      .call(d3.axisBottom(x));

    svg.append("text")
      .attr("transform", "translate(" + (width/2 + 35) + " ," + (height+1.8*margin) + ")")
      .attr("dx", "1em")
      .style("text-anchor", "middle")
      .style("font-family", "Tahoma, Geneva, sans-serif")
      .style("font-size", "12pt")
      .text(options.xLabel);

//Create the y0 axis
   var y0 = d3.scaleLinear()
    .range([height, 0])
    .domain([0, y0max + 10]);
    //.domain([0, 60]);
    //.orient("left")
svg.append("g")
    .attr("transform", "translate(" + margin + ", " + margin + ")")
    .call(d3.axisLeft(y0)
    .ticks(7));
svg.append("text")
    .attr("transform", "translate(" + 0 + " ," + ((height+2*margin)/2) + ") rotate(-90)")
    .attr("dy", "1em")
    .style("text-anchor", "middle")
    .style("font-family", "Tahoma, Geneva, sans-serif")
    .style("font-size", "12pt")
    .text(options.y0Label);

//Create the y1 axis
   var y1 = d3.scaleLinear()
    .range([height, 0])
    .domain([0, y1max + 0.1]);
    //.domain([0, 1.6]);
    //.orient("right");
svg.append("g")
    .attr("transform", "translate(" + (width+margin) + ", " + margin + ")")
    .call(d3.axisRight(y1)
    .ticks(5));
svg.append("text")
    .attr("transform", "translate(" + (width+2*margin) + " ," + ((height+2*margin)/2) + ") rotate(90)")
    .attr("dy", "1em")
    .style("text-anchor", "middle")
    .style("font-family", "Tahoma, Geneva, sans-serif")
    .style("font-size", "12pt")
    .text(options.y1Label);

//Create the chart title
svg.append("text")
    .attr("x", (width / 2 + 20))
    .attr("y", (margin/2))
    .attr("text-anchor", "middle")
    .attr("dx", "1em")
    .style("font-size", "16pt")
    .style("font-family", "Tahoma, Geneva, sans-serif")
    .text(options.plotTitle);

//Create the chart
svg.selectAll('rect')
    .data(data)
    .enter()
    .append('rect')
    .attr('y', d => (y0(d.TL) + 50))
    .attr('x', (d) => x(d.day))
    .attr('width', x.bandwidth)
    .attr('height', (d) => (height) - y0(d.TL))
    .attr('fill', options.colour)
    .attr('stroke', "#0000FF")
    .attr('stroke-width', "3");

// Add line
var line = d3.line()
        .x(function(d) { return x(d.day); })
        .y(function(d) { return y1(d.ACWR); })
       .curve(d3.curveMonotoneX);

    svg.append("path")
        .datum(data)
        .attr("class", "line")
        .attr("transform", "translate(" + 5 + "," + margin + ")")
        .attr("d", line)
        .style("fill", "none")
        .style("stroke", "black")
        .style("stroke-width", "4");

// Add legend
svg.append("circle")
  .attr("cx",margin + 20)
  .attr("cy",(margin/2 - 10))
  .attr("r", 6)
  .style("fill", "#black");

svg.append("circle")
  .attr("cx", margin + 20)
  .attr("cy",(margin/2 + 10))
  .attr("r", 6)
  .style("fill", "#87CEEB");

svg.append("text")
  .attr("x", margin + 40)
  .attr("y", (margin/2 - 10))
  .text("ACWR")
  .style("font-size", "15px")
  .attr("alignment-baseline","middle");

svg.append("text")
  .attr("x", margin + 40)
  .attr("y", (margin/2 + 10))
  .text("daily loads")
  .style("font-size", "15px")
  .attr("alignment-baseline","middle");




R/plot_ACWR.R

#' ACWR plots using d3.js
#'
#' @param db a data frame
#' @param TL training load
#' @param ACWR Acute Chronic Workload Ratio
#' @param day training days
#' @param ID ID of the subjects
#' @param colour colour of the bars. By default "#87CEEB" (skyblue)
#' @param xLabel x-axis label. By default "Days"
#' @param y0Label left y-axis label. By default "Load [AU]"
#' @param y1Label right y-axis label. By default "Acute:chronic worload ratio"
#' @param plotTitle Title of the plot. By default "ACWR"
#'
#' @import r2d3
#'
#' @return This function returns a d3.js object for a single subject.
#'          For several subjects it returns a list of d3.js objects.
#' @export
#'
#' @examples
#'
#' \dontrun{
#' # Get old working directory
#' oldwd <- getwd()
#'
#' # Set temporary directory
#' setwd(tempdir())
#'
#' # Read db
#' data("training_load", package = "ACWR")
#'
#' # Convert to data.frame
#' training_load_db <- data.frame(training_load)
#'
#' # Calculate ACWR
#' result_ACWR <- ACWR(db = training_load_db,
#'                  ID = "ID",
#'                  TL = "TL",
#'                  weeks = "Week",
#'                  days = "Day",
#'                  training_dates = "Training_Date",
#'                  ACWR_method = c("EWMA", "RAC", "RAU"))
#'
#' # Plot for 1 subject
#' # Select the first subject
#' result_ACWR_1 <- result_ACWR[result_ACWR[["ID"]] == 1,  ]
#'
#' # plot ACWR (e.g. EWMA)
#' ACWR_plot_1 <- plot_ACWR(db = result_ACWR_1,
#'                          TL = "TL",
#'                          ACWR = "EWMA_ACWR",
#'                          day = "Day")
#'
#' # Plot for several subjects
#' # plot ACWR (e.g. RAC)
#' ACWR_plot <- plot_ACWR(db = result_ACWR,
#'                          TL = "TL",
#'                          ACWR = "RAC_ACWR",
#'                          day = "Day",
#'                          ID = "ID")
#'
#' # set user working directory
#' setwd(oldwd)
#' }
#'
plot_ACWR <- function(db,
                      TL,
                      ACWR,
                      day,
                      ID = NULL,
                      colour = NULL,
                      xLabel = NULL,
                      y0Label = NULL,
                      y1Label = NULL,
                      plotTitle = NULL) {

  # Check variables
  if(is.null(db)){
    stop("you must provide a db")
  }
  if(is.null(TL)){
    stop("you must provide the name of the training load column in the database")
  }
  if(is.null(ACWR)){
    stop("you must provide the name of the ACWR column in the database")
  }
  if(is.null(day)){
    stop("you must provide the name of the day training column in the database")
  }
  if(is.null(colour)){
    colour <- "#87CEEB"
  }
  if(is.null(xLabel)){
    xLabel <- "Days"
  }
  if(is.null(y0Label)){
    y0Label <- "Load [AU]"
  }
  if(is.null(y1Label)){
    y1Label <- "Acute:chronic worload ratio"
  }
  if(is.null(plotTitle)){
    plotTitle <- "ACWR"
  }

  # Rename db columns
  # TL
  names(db)[names(db) == TL] <- "TL"
  # ACWR
  names(db)[names(db) == ACWR] <- "ACWR"
  # day
  names(db)[names(db) == day] <- "day"

  # Single plot
  if(is.null(ID)){

  # Day / TL / EWMA_ACWR
  d3_ACWR <- r2d3(data = db,
             script = system.file("ACWR_plot.js", package = "ACWR"),
             options = list(margin = 50,
                            #barPadding = 0.1,
                            colour = colour,
                            xLabel = xLabel,
                            y0Label = y0Label,
                            y1Label = y1Label,
                            plotTitle = plotTitle
             ),
  )

  # return d3 object
  return(d3_ACWR)

  } # end if is.null(ID)

  # Multiple plots
  if(!is.null(ID)){

    # Create an empty list to store the plots
    d3_ACWR_list <- list()

    # Loop over the subjects
    for (i in unique(db[[ID]])) {

      # Create individual dfs
      db_ind <- db[db[[ID]] == i,  c("TL", "ACWR", "day") ]

      # Individual plots
      d3_ACWR <- r2d3(data = db_ind,
                      script = system.file("ACWR_plot.js", package = "ACWR"),
                      options = list(margin = 50,
                                     #barPadding = 0.1,
                                     colour = colour,
                                     xLabel = xLabel,
                                     y0Label = y0Label,
                                     y1Label = y1Label,
                                     plotTitle = plotTitle
                      ),
      )

    # Store the plots inside the list
    d3_ACWR_list[[paste0("ID = ", i)]] <- d3_ACWR

    }

    # return a list
    return(d3_ACWR_list)
  } # end if !is.null(ID)


}
